lib/options_validator.js

```diff
@@ -209,23 +209,25 @@ class OperationBuilder {
         validationOptions = { validationLevel: this.s.options.validationLevel };
       }
 
-      // validate options
-      const optionsIndex =
-        typeof args[args.length - 1] === 'function' ? args.length - 2 : args.length - 1;
+      const callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;
+      const options = typeof args[operationBuilder.requiredArity] === 'object' ? args.pop(): {};
```

Comment:
In this test case:

```js
const fn = arityOne().options({...}).build(...);
fn({a: 1}, {b: 2}, () => {}, callback);
```

This line will confirm `typeof {b: 2} === 'object'`, but then pop `() => {}` as the options object. I want to double check: will `assertArity` catch this, and warn before this happens?

Reply:
@kvwalker let's answer this with a unit test!